### ISSUE TITLE
Shell completion

### ISSUE THREAD

**Issue by djmattyg007** [dmerejkowsky/ruplacer#97]
It would be great to have support for shell completion for bash, zsh, etc.

**Comment by dmerejkowsky** [comment]
Yeah, but I once tried to write a completion function by hand and hated every minute of it :stuck_out_tongue: 

Also, I prefer no auto completion at all rather than a faulty one but maybe that's just me.

Feel free to try and send a patch, though ;)

**Comment by djmattyg007** [comment]
What about integrating clap-complete?

https://github.com/clap-rs/clap/tree/master/clap_complete

**Comment by dmerejkowsky** [comment]
Dunno. Might work. Also, one of the links in clap complete's talks about `fig`, which seems the least painful way to write completion scripts I've seen so far ...

Might try it one day. Again, feel free to try yourself :)
